src/libYARP_serversql/src/yarpserver.cpp

```diff
@@ -174,41 +176,32 @@ class NameServerContainer : public ComposedNameService {
     }
 };
 
-yarpserversql_API yarp::os::NameStore *yarpserver_create(yarp::os::Searchable& options) {
-    NameServerContainer *nc = new NameServerContainer;
-    if (!nc) return nullptr;
-    nc->setSilent(true);
-    if (!nc->open(options)) {
-        delete nc;
-        return nullptr;
-    }
-    nc->goPublic();
-    return nc;
-}
-
-yarpserversql_API int yarpserver_main(int argc, char *argv[]) {
+void YarpServerThread::run()
+{
     // check if YARP version is sufficiently up to date - there was
     // an important bug fix
-    bool silent(false);
-    Bottle b("ip 10.0.0.10");
-    if (b.get(1).asString()!="10.0.0.10") {
-        fprintf(stderr, "Sorry, please update YARP version");
-        ::exit(1);
-    }
 
-    Property options;
+    Property            options;
+    bool                silent(false);
+    FILE*               out;
+    NameServerContainer nc; if (!nc.open(options)) return;
```

Comment:
I would move this line after the next if-else block, to change as little as possible.